src/mesh_analysis/tracers/waku_tracer.py

```diff
@@ -15,21 +20,108 @@ def __init__(self):
         # - Different patterns (received, sent, dropped)
         # - Once one pattern search is completed, stop search for it in the logs (ie: Announce Address)
         super().__init__()
-        self._patterns = [r'my_peer_id=([\w*]+).*?msg_hash=(0x[\da-f]+).*?receivedTime=(\d+)']
+        self._patterns = []
+        self._tracings = []
 
-    def trace(self, parsed_logs: List) -> pd.DataFrame:
-        df = self._trace_message_in_logs(parsed_logs)
+    def with_received_pattern(self):
+        self._patterns.append(
+            r'received relay message.*?my_peer_id=([\w*]+).*?msg_hash=(0x[\da-f]+).*?from_peer_id=([\w*]+).*?receivedTime=(\d+)')
+        self._tracings.append(self._trace_received_in_logs)
 
-        return df
+    def with_sent_pattern(self):
+        self._patterns.append(
+            r'sent relay message.*?my_peer_id=([\w*]+).*?msg_hash=(0x[\da-f]+).*?to_peer_id=([\w*]+).*?sentTime=(\d+)')
+        self._tracings.append(self._trace_sent_in_logs)
+
+    def with_wildcard_pattern(self):
+        self._patterns.append(r'(.*)')
+        self._tracings.append(self._trace_all_logs)
+
+    def trace(self, parsed_logs: List) -> List:
+        dfs = []
+        for i, trace in enumerate(self._tracings):
+            if trace is not None:
+                dfs.append(trace(parsed_logs[i]))
+
+        return dfs
```

Comment:
I'd suggest using a list comprehension for this for idiomacy and consistency.

Reply:
Yup yup!